src/engine/structures.rs

```diff
@@ -0,0 +1,156 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+use std::collections::HashMap;
+use std::slice::IterMut;
+
+use uuid::Uuid;
+
+use super::engine::{HasName, HasUuid};
+
+
+/// Map UUID and name to T items.
+#[derive(Debug)]
+pub struct Table<T: HasName + HasUuid> {
+    items: Vec<T>,
+    name_map: HashMap<String, usize>,
+    uuid_map: HashMap<Uuid, usize>,
+}
```

Comment:
I think I've answered that in my Table commit message. Essentially, there are lots of benefits to encapsulation that it is worthwhile realizing here. I considered a variety of two-map approaches. None of them were as consistent or clear as the current approach. Working through what is necessary to remove an entry given a UUID and then working through what it necessary to find one given a name will show that these two-map approaches are surprisingly hard to support. A bidirectional map would ease things, but the one we have been using is both O(n), which could become a problem, and buggy (in its insertion function). I would _much_ rather use standard library data structures.